packages/sdk/src/solana/programs/nft-collection.ts

```diff
@@ -175,54 +179,16 @@ export class NFTCollection {
    * ```
    */
   async supplyOf(nftAddress: string): Promise<bigint> {
-    let editionMarkerNumber = 0;
-    let totalSupply = 1;
+    const originalEditionAccount = await this.metaplex
+      .rpc()
+      .getAccount(findMasterEditionV2Pda(new PublicKey(nftAddress)));
 
-    cursedBitwiseLogicLoop: while (true) {
-      const editionMarkerAddress = findEditionMarkerPda(
-        new PublicKey(nftAddress),
-        toBigNumber(editionMarkerNumber * 248),
-      );
-      const editionMarker = await EditionMarker.fromAccountAddress(
-        this.metaplex.connection,
-        editionMarkerAddress,
-      );
-
-      // WARNING: Ugly bitwise operations because of Rust :(
-      const indexCap = editionMarkerNumber === 0 ? 247 : 248;
-      for (let editionIndex = 0; editionIndex < indexCap; editionIndex++) {
-        const ledgerIndex =
-          editionMarkerNumber > 0
-            ? Math.floor(editionIndex / 8)
-            : editionIndex < 7
-            ? 0
-            : Math.floor((editionIndex - 7) / 8) + 1;
-        const size = editionMarkerNumber === 0 && ledgerIndex === 0 ? 7 : 8;
-        const shiftBase = 0b1 << (size - 1);
-
-        const bitmask =
-          ledgerIndex === 0
-            ? shiftBase >> editionIndex
-            : editionMarkerNumber > 0
-            ? shiftBase >> editionIndex % (ledgerIndex * 8)
-            : ledgerIndex === 1
-            ? shiftBase >> (editionIndex - 7)
-            : shiftBase >> (editionIndex - 7) % ((ledgerIndex - 1) * 8);
-
-        const editionExists =
-          (editionMarker.ledger[ledgerIndex] & bitmask) !== 0;
-
-        if (editionExists) {
-          totalSupply += 1;
-        } else {
-          break cursedBitwiseLogicLoop;
-        }
-      }
+    const originalEdition = toNftOriginalEdition(
+      toOriginalEditionAccount(originalEditionAccount),
+    );
 
-      editionMarkerNumber++;
-    }
-
-    return BigInt(totalSupply);
+    // Add one to supply to account for the master edition
```

Comment:
I think we need to check if the token actually exists too, could have been burned so it's not always guaranteed to have supply 1